### DISCUSSION TITLE
Tooltip extension

### DISCUSSION THREAD

**Discussion by thefinnomenon** [ueberdosis/tiptap#2603] [General Questions]
I am trying to create a tooltip extension where you can select some text and add a mark (something like <span data-type='tooltip' content="..." /> and on hover, it shows the editable tooltip. Essentially I think I need to render an editable node view in a Tippy instance that is linked to the span. I am having trouble getting this to work though and would appreciate any suggestions.

**Comment by rfgamaral** [comment]
Sorry if I misunderstood, but doesn't the [BubbleMenu](https://tiptap.dev/api/extensions/bubble-menu) solve your issue?

**Reply by thefinnomenon** [reply]
I am using the BubbleMenu set to show when the tooltip mark (<span data-type="tooltip">) is hovered and I can add an input field to it to be able to set plain text for the tooltip but I really want it to accept rich text so the user can enter enter header text, make some text bold, etc. I did this with a callout block using a node view but can't figure out how I would replicate this in this case. I am thinking I am going to have to do something like the mention & suggestion extensions but wanted to ask here first to see if there is a simpler way.
